librz/include/rz_util/rz_print.h

```diff
@@ -41,6 +42,49 @@ typedef const char *(*RzPrintSectionGet)(void *user, ut64 addr);
 typedef const char *(*RzPrintColorFor)(void *user, ut64 addr, bool verbose);
 typedef char *(*RzPrintHasRefs)(void *user, ut64 addr, int mode);
 
+typedef enum {
+	RZ_ASM_TOKEN_NONE, // Type which does not fit any else.
+	RZ_ASM_TOKEN_MNEMONIC, // Asm mnemonics like: mov, push, lea...
+	RZ_ASM_TOKEN_OPERATOR, // Arithmetic operators: +,-,<< etc.
+	RZ_ASM_TOKEN_NUMBER, // Numbers
+	RZ_ASM_TOKEN_REGISTER, // Registers
+	RZ_ASM_TOKEN_SEPARATOR, // Brackets, comma etc.
+	RZ_ASM_TOKEN_META, // Meta information (e.g Hexagon packet prefix, ARM & Hexagon number prefix).
+	RZ_ASM_TOKEN_UNKNOWN, // Does not fit to any token above.
+} RzAsmTokenType;
+
+/**
+ *  \brief A token of an asm string holding meta data.
+ */
+typedef struct {
+	size_t start; // byte-offset into `str` where this token starts. Must be exactly at a utf-8 codepoint boundary.
+	size_t len; // `str` length of token in bytes.
+	RzAsmTokenType type;
+	union {
+		ut64 number; // Number of RZ_ASM_TOKEN_NUMBER
+		ut16 reg_num; // Register number
```

Comment:
what is a `register number` ?

Reply:
In case the analysis plugin knows the number of the general register, it can enter it here. E.g. R0 = 0, R31 = 31 etc.
Not sure whether it is useful.

Reply:
i suggest to replace it with a string pointer

Reply:
The idea was to allow easy recognition of flag values in the future. So in case those tokens are used for more than coloring (e.g. replacing numbers with the flags) the comparison would not be against strings.